### DISCUSSION TITLE
Tailwind Theme Customization Only Extend

### DISCUSSION THREAD

**Discussion by browner12** [tailwindlabs/tailwindcss#6600] [Ideas]
Shouldn't all theme customization in `tailwind.config.js` just essentially be "extending" now?

Prior to the JIT, "extending" allowed you to add or override the defaults, while also keeping and generating all the default classes.

"Non-extending" would be used if you specifically wanted to essentially get rid of some options, and completely replace them with your own options.

With the JIT, since we're only building classes that are actually in use, can't we just **always** extend our theme, and if it's a new key obviously add it, and if it's an existing key, overwrite the default?

https://v2.tailwindcss.com/docs/theme

**Comment by adamwathan** [comment]
The main arguments against this I think are autocompletions + just wanting to have tighter control over what values are available to make sure everything matches your design system.

You don't want `shadow-xl` to show up in your autocomplete suggestions (or actually work if someone uses it) if your design system just contains `shadow` and `shadow-lg` you know?

**Reply by browner12** [reply]
Interesting perspective, yah this makes sense.  Thanks.
